Update rendering plugins for TB2025

ywkao opened the pull request:
#### Summary
Updates DQM rendering plugins for the 2025 testbeam campaign with HGCAL-focused enhancements and improved visualization capabilities. The plugins are deployed on `hgc-vm-2025`.

#### Key Changes
- Custom color palettes for quality histograms and fixed axis label shifting in 2D plots
- Add under/overflow info for TH1F histograms
- Set rendering effect for BX/L1A/Orbit 2D validation plots

#### Note
The branch will keep updating till the end of the 2025 testbeam campaign (08 Oct. 2025).

ywkao committed:
Set quality color code & some tweaks

ywkao committed:
Enable FED and Quick directories & Tweak canvas margins

ywkao committed:
feat: improve HGCAL quality histogram rendering with custom color palette

- Add custom 5-color palette for econdQuality histograms (green to red)
- Set appropriate min/max range (0.5-5.5) for quality grade scale
- Optimize rendering logic to prevent duplicate palette settings
- Improve code maintainability with clearer conditional logic

ywkao committed:
update layouts for 2025 DQM demo

ywkao committed:
Disable axis center labels to fix label-shifting in HGCAL 2D plots

ywkao committed:
Keep only HGCAL-related rendering plugins

ywkao committed:
update rendering effects for BX/L1A/Orbit 2D validation plots

ywkao committed:
add under/overflow info for TH1F histograms

ywkao committed:
update plots in Layouts and QuickCollection folders

ywkao committed:
remove timestamp from the layout/quick collection folders

ywkao committed:
display adc wafer maps in the layout folder

ywkao committed:
improve figure names in the layout folder

ywkao committed:
ensure TH1F objects drawn with 'hist' option

ywkao committed:
use inverted kcherry for 2D plots

ywkao committed:
use inverted kSunset & add debug message

ywkao committed:
restrict noise scale [0, 2] and update module at layer level

ywkao committed:
show text info for MEs at layer level

ywkao committed:
add issue severity title to quality MEs

ywkao committed:
customize markers for overflow channels

ywkao committed:
add trig phase in the layout folder

ywkao committed:
put plots under Layouts/trigPhase

ywkao committed:
add ADC, ToT, ToA folders under Layouts

ywkao committed:
add a profile to TH2F

ywkao committed:
enable overlaying a profile on a TH2F

ywkao committed:
add RecHit layout folder & zoom in hexagonal plots

ywkao committed:
remove profile from econ-d quality th2f

Commit:
streamline usage of module typecodes

pfs committed:
Update hgcal-layouts.py

ywkao committed:
convert avgadc profile to th1d

ywkao committed:
fix compilation error

pfs commented:
Hi Yu-Wei, thanks, did you get my PR from yesterday night? 
https://github.com/ywkao/hgc-dqmgui/pull/1
It seems to me that when constructing the `hgcal-layouts.py` file we can do it in a more programatic way instead of writing one line per layer or type code.
Could this be done before integrating?
Unfortunately for some reason when trying to deploy it yesterday night, I could not do it adequately. But I don't see no reason why it would not work, and it would simplify greatly the maintenance for future test systems.

ywkao committed:
adapt to TB backup Si setup2

ywkao committed:
remove profile for TH2F associated with (econd) quality

ywkao committed:
Merge branch 'dev-tb2025' into dev-tb2025

ywkao committed:
Merge pull request #1 from CMS-HGCAL/dev-tb2025

streamline usage of module typecodes

ywkao committed:
replace lines with for-loops

ywkao committed:
streamlined hgcal-layouts.py

ywkao commented:
Hi @pfs, thank you for the suggestion! The code in `hgcal-layout.py` has been further simplified using string templates. Let me know what you think about the update.

ywkao commented:
The change has been tested on `hgc-vm-2025.cern.ch`. It works fine.

pfs commented:
thank you!

ywkao committed:
Fix rendering order: prevent TH2 styling from overriding TH2Poly plots

ywkao committed:
Remove TB2025-specific zoom functionality

The zoom-in feature was added specifically for TB2025 but
should not be in the main branch as it affects all use cases
and would complicate future debugging.

ywkao committed:
code cleaning

ywkao commented:
@pfs, ready to merge. Removed TB2025 zoom to keep the main branch generic.

pfs commented:
thank you!
